### PR TITLE
Allow the use of remote ruleset files that are hosted on http

### LINKED ISSUE #149: Allow the use of remote ruleset files that are hosted on http
**Describe the bug**
It looks like its is not possible to use custom rulesets that are hosted on HTTP such as: https://api.oneadvanced.com/rules/spectral.yaml instead of having a local rule set file.
**To Reproduce**

1. Update the VS code extension setting "Spectral Ruleset File" to be a remote ruleset
3. Open an API and create a query parameter that is not cased in camelCase

**Expected behavior**
The custom ruleset is loaded and an error is displayed. (This is what happens with the CLI today)

**Environment:**
 - Extension version: [0.2.5]
 - VS Code version: [1.6.8]
 - Operating System: [10]

### PR DESCRIPTION
**Pull request by NickHeap2**
Fixes #149
Remote rulesets can now be used.

**Checklist**

- [X] Tests added / updated
- [X] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

### CODE REVIEW

**Review by P0lip — APPROVED** [review]
LGTM. Thanks a lot.

---

FWIW, do feel free to apply the following change, so that you can run things on Windows without any changes. 

```diff
diff --git a/make.js b/make.js
index 82fb2e6..f7c888a 100755
--- a/make.js
+++ b/make.js
@@ -36,7 +36,7 @@ target.package = async () => {
   await buildAll();
 
   pushd(outputPath.dist);
-  run(`node ../node_modules/.bin/vsce package -o ${outputPath.artifacts}`);
+  run(`node ${path.join(__dirname, './node_modules/.bin/vsce')} package -o ${outputPath.artifacts}`);
   popd();
 };

```

File: server/src/server.ts [hunk 1]
```diff
@@ -211,11 +218,15 @@ function resolveSettings(document: TextDocument): Thenable<TextDocumentSettings>
...
+          connection.sendNotification(StartWatcherNotification.type, { path: rulesetFile });
+        }
         try {
           const { ruleset, dependencies } = await Linter.loadRuleset(rulesetFile, { fs, fetch });
-          for (const dependency of dependencies) {
-            connection.sendNotification(StartWatcherNotification.type, { path: dependency });
+          if (!rulesetFileIsUrl) {
+            for (const dependency of dependencies) {
+              connection.sendNotification(StartWatcherNotification.type, { path: dependency });
+            }
```
Comment (P0lip) [inline server/src/server.ts:229]: ^
```suggestion
          for (const dependency of dependencies) {
            connection.sendNotification(StartWatcherNotification.type, { path: dependency });
          }
```

**Comment by NickHeap2** [comment]
@P0lip
``run(`node ${path.join(__dirname, './node_modules/.bin/vsce')} package -o ${outputPath.artifacts}`);``

I can only get it to run on windows if I remove the node part:

``run(`${path.join(__dirname, './node_modules/.bin/vsce')} package -o ${outputPath.artifacts}`);``

File: server/src/server.ts [hunk 2]
```diff
@@ -211,11 +218,15 @@ function resolveSettings(document: TextDocument): Thenable<TextDocumentSettings>
...
 
-        connection.sendNotification(StartWatcherNotification.type, { path: rulesetFile });
+        if (!rulesetFileIsUrl) {
+          connection.sendNotification(StartWatcherNotification.type, { path: rulesetFile });
+        }
         try {
           const { ruleset, dependencies } = await Linter.loadRuleset(rulesetFile, { fs, fetch });
-          for (const dependency of dependencies) {
-            connection.sendNotification(StartWatcherNotification.type, { path: dependency });
+          if (!rulesetFileIsUrl) {
```
[the remark this answers is not in the document]
Comment (NickHeap2) [reply server/src/server.ts]: Done

File: package.json [hunk 3]
```diff
@@ -19,7 +19,7 @@
           "type": "boolean"
         },
         "spectral.rulesetFile": {
-          "description": "Location of the ruleset file to use when validating. If omitted, the default is a .spectral.yml/.spectral.json in the same folder as the document being validated. Paths are relative to the workspace.",
+          "description": "Location of the ruleset file to use when validating. If omitted, the default is a .spectral.yml/.spectral.json in the same folder as the document being validated. Paths are relative to the workspace. This can also be a remote url.",
```
[the remark this answers is not in the document]
Comment (NickHeap2) [reply package.json]: Done

File: client/src/__tests__/e2e/helper.ts [hunk 4]
```diff
@@ -14,6 +14,15 @@ export const activate = async (): Promise<void> => {
   console.info(`Extension '${ext.id}' v${ext.packageJSON.version} has been successfully activated.`);
 };
 
+export const setRulesetFile = (rulesetFile: string): void => {
+  try {
+    vscode.workspace.getConfiguration('spectral')
+      .update('rulesetFile', rulesetFile, true, false);
+  } catch (error) {
+    console.log(error);
+  }
```
[the remark this answers is not in the document]
Comment (NickHeap2) [reply client/src/__tests__/e2e/helper.ts]: Done

File: server/src/server.ts (hunk 1 above)
Comment (NickHeap2) [reply server/src/server.ts]: Done

**Comment by P0lip** [comment]
> I can only get it to run on windows if I remove the node part:

I'm on Mac, and the the latter works on my end either, so let's stick with it.

**Comment by NickHeap2** [comment]
> > I can only get it to run on windows if I remove the node part:
> 
> I'm on Mac, and the the latter works on my end either, so let's stick with it.

OK that's all the suggestions done.

**Review by P0lip — APPROVED** [review]
Thank you!

**Pull request merged**